src/connectors/impls/gcs/writer.rs

```diff
@@ -0,0 +1,831 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::connectors::prelude::{
+    Attempt, ErrorKind, EventSerializer, Result, SinkAddr, SinkContext, SinkManagerBuilder,
+    SinkReply, Url,
+};
+use crate::connectors::sink::Sink;
+use crate::connectors::utils::url::HttpsDefaults;
+use crate::connectors::{
+    CodecReq, Connector, ConnectorBuilder, ConnectorConfig, ConnectorType, Context,
+};
+use crate::system::KillSwitch;
+use crate::{connectors, QSIZE};
+use async_std::channel::{bounded, Receiver, Sender};
+use async_std::task::sleep;
+#[cfg(not(test))]
+use gouth::Token;
+#[cfg(not(test))]
+use http_client::h1::H1Client;
+use http_client::HttpClient;
+use http_types::{Method, Request};
+use std::collections::HashMap;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+#[cfg(test)]
+use tests::MockHttpClient;
+use tremor_pipeline::{ConfigImpl, Event};
+use tremor_value::Value;
+use value_trait::ValueAccess;
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct Config {
+    #[serde(default = "default_endpoint")]
+    endpoint: String,
```

Comment:
we generalized to `url`  for everything that takes  url I think we should do it here too and not use `String` but `Url::<HttpsDefaults>` as a type.

Reply:
ohhh, right, I didn't think to even try deserializing this, but it's a good idea, will make the code a bit simpler :)